File: problems/problems_2125/solution.go
package problem2125

import (
	"encoding/json"
	"log"
	"strings"
)

func numberOfBeams(bank []string) (ans int) {
	prev := 0
	for _, row := range bank {
		count := 0
		for _, b := range row {
			if b == '1' {
				count++
			}
		}
		ans += count * prev
		if count > 0 {
			prev = count
		}
	}
	return
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var bank []string

	if err := json.Unmarshal([]byte(inputValues[0]), &bank); err != nil {
		log.Fatal(err)
	}

	return numberOfBeams(bank)
}

File: problems/problems_2125/Solution.java
package problems.problems_2125;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int numberOfBeams(String[] bank) {
        int ans = 0;
        int prev = 0;
        for (String row: bank) {
            int count = 0;
            for (int i = 0; i < row.length(); i++) {
                if (row.charAt(i) == '1') {
                    count++;
                }
            }
            if (count > 0) {
                ans += prev * count;
                prev = count;
            }
        }
        return ans;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        String[] bank = jsonArrayToStringArray(inputJsonValues[0]);
        return JSON.toJSON(numberOfBeams(bank));
    }
}
